snowflake/data_access.go

```diff
@@ -233,35 +226,57 @@ func (s *AccessSyncer) SyncAccessAsCodeToTarget(ctx context.Context, access map[
 		return err
 	}
 
-	rolesToRemove := make([]string, 0)
+	rolesToRemove := make(map[string]*importer.AccessProvider, 0)
 
 	for role, toKeep := range existingRoles {
 		if !toKeep {
-			rolesToRemove = append(rolesToRemove, role)
+			rolesToRemove[role] = access[role]
 		}
 	}
 
-	err = s.removeRolesToRemove(rolesToRemove, repo)
+	err = s.removeRolesToRemove(rolesToRemove, repo, nil)
```

Comment:
I was thinking... Shouldn't we create a dummy feedbackHandler. Someone that can be used but do nothing at all? In that cases there is no reason to everywhere check if this is nil or not